jcl/src/java.base/share/classes/java/lang/Thread.java

```diff
@@ -1488,6 +1489,38 @@ public static enum State {
 	 */
 	TERMINATED }
 
+/**
+ * Returns the translation from a J9VMThread state to a Thread::State.
+ *
+ * @return this thread's state.
+ *
+ * @see State
+ */
+private State translateJ9VMThreadStateToThreadState(int status) {
+	switch (status) {
+		case 1: // J9VMTHREAD_STATE_RUNNING
+			return State.RUNNABLE;
+		case 2: // J9VMTHREAD_STATE_BLOCKED
+			return State.BLOCKED;
+		case 4: // J9VMTHREAD_STATE_WAITING
+		case 0x80: // J9VMTHREAD_STATE_PARKED
+			return State.WAITING;
+		case 8: // J9VMTHREAD_STATE_SLEEPING
+		case 64: // J9VMTHREAD_STATE_WAITING_TIMED
+		case 0x100: // J9VMTHREAD_STATE_PARKED_TIMED
+			return State.TIMED_WAITING;
+		case 32: // J9VMTHREAD_STATE_DEAD
+			return State.TERMINATED;
+		default:
+			synchronized (lock) {
+				if (threadRef == NO_REF) {
+					return State.TERMINATED;
+				}
+				return State.values()[getStateImpl(threadRef)];
+			}
+	}
```

Comment:
Ref: https://github.com/eclipse-omr/omr/blob/master/doc/CodingStandard.md#formatting-switch-statements

Switch labels (`case` or `default`) are indented to the same level as the `switch` keyword.

```suggestion
	switch (status) {
	case 1: // J9VMTHREAD_STATE_RUNNING
		return State.RUNNABLE;
	case 2: // J9VMTHREAD_STATE_BLOCKED
		return State.BLOCKED;
	case 4: // J9VMTHREAD_STATE_WAITING
	case 0x80: // J9VMTHREAD_STATE_PARKED
		return State.WAITING;
	case 8: // J9VMTHREAD_STATE_SLEEPING
	case 64: // J9VMTHREAD_STATE_WAITING_TIMED
	case 0x100: // J9VMTHREAD_STATE_PARKED_TIMED
		return State.TIMED_WAITING;
	case 32: // J9VMTHREAD_STATE_DEAD
		return State.TERMINATED;
	default:
		synchronized (lock) {
			if (threadRef == NO_REF) {
				return State.TERMINATED;
			}
			return State.values()[getStateImpl(threadRef)];
		}
	}
```